File: SEP3_Database/src/main/java/com/example/SEP3_Database/model/domain/unit/message/Message.java
package com.example.SEP3_Database.model.domain.unit.message;

import com.example.SEP3_Database.model.domain.unit.time.MyTime;
import java.time.LocalDateTime;


public abstract class Message {
    public String MessageInfo;
    public MessageType Type;
    public String SenderId;
    public MyTime Time;
    public Message(String senderId,String messageInfo,MessageType type)
    {
        SenderId = senderId;
        MessageInfo = messageInfo;
        Type = type;
        Time = new MyTime(LocalDateTime.now());
    }

    public Message(String senderId,String messageInfo,MessageType type,MyTime time)
    {
        SenderId = senderId;
        MessageInfo = messageInfo;
        Type = type;
        Time = time;
    }

    public String getMessageInfo() {
        return MessageInfo;
    }

    public void setMessageInfo(String messageInfo) {
        MessageInfo = messageInfo;
    }

    public MessageType getMessageType() {
        return Type;
    }

    public void setMessageType(MessageType type) {
        Type = type;
    }

    public String getSenderId() {
        return SenderId;
    }

    public void setSenderId(String senderId) {
        SenderId = senderId;
    }

    public MyTime getTime() {
        return Time;
    }

    public void setTime(MyTime time) {
        Time = time;
    }
}

File: SEP3_Client/Model/Unit/Message/Message.cs
using System;
using System.Text.Json.Serialization;
using SEP3_Client.Model.Unit.Time;

namespace SEP3_Client.Model.Unit.Message
{
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public abstract class Message
    {
        public string MessageInfo { get; set; }
        public MessageType Type { get; set; }
        public string SenderId { get; set; }
        public MyTime Time { get; set; }
        public Message(string senderId,string messageInfo,MessageType type)
        {
            SenderId = senderId;
            MessageInfo = messageInfo;
            Type = type;
            Time = new MyTime(DateTime.Now);
        }
        
        public Message() {}

        public abstract bool CanRead(string userId);
    }
}